src/extension.ts

```diff
@@ -240,6 +240,19 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         );
       });
     }),
+    registerCommand(`${serverId}.printDebugInformation`, async () => {
+      if (!lsClient) {
```

Comment:
Can we only register the command if this is the new LSP server because I suspect it isn't supported for `ruff-lsp`?

Reply:
Yeah, I agree. `ruff-lsp` doesn't have the `ruff.printDebugInformation` command in its capabilities so it'll throw an error.

Reply:
Confirmed that it'll raise an exception:
```
Failed to handle request 2 workspace/executeCommand ExecuteCommandParams(command='ruff.printDebugInformation', arguments=[{'uri': 'file:///Users/dhruv/playground/ruff/src/TCH002.py'}], work_done_token=None)
Traceback (most recent call last):
  File "/Users/dhruv/.local/state/pipx/venvs/ruff-lsp/lib/python3.12/site-packages/pygls/protocol/json_rpc.py", line 264, in _handle_request
    handler(params, msg_id)
  File "/Users/dhruv/.local/state/pipx/venvs/ruff-lsp/lib/python3.12/site-packages/pygls/protocol/lsp_meta.py", line 17, in decorator
    ret_val = base_func(self, *args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/dhruv/.local/state/pipx/venvs/ruff-lsp/lib/python3.12/site-packages/pygls/protocol/language_server.py", line 320, in lsp_workspace__execute_command
    cmd_handler = self.fm.commands[params.command]
                  ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
KeyError: 'ruff.printDebugInformation'
```